src/core/etl/src/Flow/ETL/PHP/Type/ArrayContentDetector.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Flow\ETL\PHP\Type;
+
+use Flow\ETL\Exception\InvalidArgumentException;
+use Flow\ETL\PHP\Type\Native\ArrayType;
+use Flow\ETL\PHP\Type\Native\NullType;
+use Flow\ETL\PHP\Type\Native\ScalarType;
+
+final class ArrayContentDetector
+{
+    private readonly ?Type $firstKeyType;
+
+    private readonly ?Type $firstValueType;
+
+    public function __construct(private readonly Types $uniqueKeysType, private readonly Types $uniqueValuesType)
+    {
+        $this->firstKeyType = $this->uniqueKeysType->first();
+        $this->firstValueType = $this->uniqueValuesType->first();
```

Comment:
You can just cache it in the Types, and here just return `$this->uniqueValuesType->first()` when it's needed